driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java

```diff
@@ -111,8 +142,31 @@ public BsonDocument getResumeToken() {
      *
      * @return the namespace
      */
+    @BsonIgnore @Nullable
     public MongoNamespace getNamespace() {
-        return namespace;
+        return namespaceDocument != null
+                ? new MongoNamespace(namespaceDocument.getString("db").getValue(), namespaceDocument.getString("coll").getValue())
+                : null;
+    }
+
+    /**
+     * Returns the namespaceDocument
```

Comment:
Indicate what the expected contents of the document are.

Reply:
Done